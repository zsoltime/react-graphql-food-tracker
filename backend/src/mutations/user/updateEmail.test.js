'use strict';

const updateEmail = require('./updateEmail');

describe('updateEmail mutation', () => {
  it('sets user a pending email and returns a message', async () => {
    const args = { email: 'test@email.io' };
    const mockedContext = {
      db: {
        mutation: { updateUser: jest.fn() },
        query: {
          user: jest
            .fn()
            .mockImplementation(() => Promise.resolve(null)),
        },
      },
      request: { userId: 'testUserId' },
    };
    const res = await updateEmail(null, args, mockedContext);

    expect(res).toMatchSnapshot();
    expect(
      mockedContext.db.query.user.mock.calls[0]
    ).toMatchSnapshot();
    expect(mockedContext.db.mutation.updateUser).toHaveBeenCalledWith(
      expect.objectContaining({
        data: {
          pendingEmail: args.email,
          token: expect.any(String),
          tokenExpiry: expect.any(Number),
        },
        where: { id: 'testUserId' },
      })
    );
  });

  it('throws error if user not signed in', () => {
    const mockedContext = {
      request: {},
    };

    expect(
      updateEmail(null, null, mockedContext)
    ).rejects.toThrowErrorMatchingSnapshot();
  });

  it('throws error if email is invalid', () => {
    const args = { email: 'invalidemail.io' };
    const mockedContext = {
      request: { userId: 'testUserId' },
    };

    expect(
      updateEmail(null, args, mockedContext)
    ).rejects.toThrowErrorMatchingSnapshot();
  });

  it('throws error if email is already in use', () => {
    const args = { email: 'valid@email.io' };
    const mockedContext = {
      db: {
        query: {
          user: jest
            .fn()
            .mockImplementation(() => Promise.resolve({})),
        },
      },
      request: { userId: 'testUserId' },
    };

    expect(
      updateEmail(null, args, mockedContext)
    ).rejects.toThrowErrorMatchingSnapshot();
  });
});
