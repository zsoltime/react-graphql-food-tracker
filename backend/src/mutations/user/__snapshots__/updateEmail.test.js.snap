// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`updateEmail mutation sets user a pending email and returns a message 1`] = `
Object {
  "message": "Email with token sent",
}
`;

exports[`updateEmail mutation sets user a pending email and returns a message 2`] = `
Array [
  Object {
    "where": Object {
      "email": "test@email.io",
    },
  },
]
`;

exports[`updateEmail mutation throws error if email is already in use 1`] = `"Another user is already using this email"`;

exports[`updateEmail mutation throws error if email is invalid 1`] = `"Email is not valid"`;

exports[`updateEmail mutation throws error if user not signed in 1`] = `"You must be signed in to update your email address"`;
